Automattic/wp-calypso
# Editor: Enable Drag/Drop for Featured Image component in the editor

This PR enables Drag/Drop functionality over the Featured Image component in the sidebar of the editor as mentioned in #324.

The scope of this PR is just to add the DropZone to the sidebar accordion and allow to upload a featured image through there. Future improvements to the feature and adding additional dropzones will be targets of other, separate, PRs.

How it looks as of f64fa15:
![fi-dnd](https://cloud.githubusercontent.com/assets/184938/24873029/1cd337e0-1e28-11e7-8110-4cea12ad6543.gif)

previous versions: [#](https://cloud.githubusercontent.com/assets/184938/24051725/b2b48f48-0b3b-11e7-81cb-125b1dc8f2a6.gif)

To test:
1. Checkout branch or use Calypso.live link
2. Start editing a post or create a new post
3. Drag an image over the featured image block in the sidebar
4. Verify the dropzone shows 
5. Verify that the dropped image is set as the featured image
6. Verify that saving the post would retain the featured image
7. Verify that there are no JS errors in the console
8. Verify that you can still upload a featured image when clicking the button (old way)
9. Verify you can upload image on a mobile device (touch)
